CONTRIBUTING.md

```diff
@@ -12,3 +12,7 @@ Come hang out at https://github.com/a-b-street/osm2streets to ask questions and
 See the `tests` crate [(docs)](tests/README.md) for a collection of interesting example scenarios.
 
 See `street-explorer` crate [(docs)](street-explorer/README.md) for a *Street Explorer* test browsing interface.
+
+## Design decisions
+
+The `Road` and `Intersection` structs are rather stateful. `trimmed_road_line` and `polygon` are derived state, produced by a transformation step. Derived state isn't updated automatically, so you have to be internally careful about knowing what transformations have run. Some earlier ones call `estimate_trimmed_geometry` for this reason.
```

Comment:
If we wanted to make it safer to use `trimmed_road_line` and `polygon` without paying the cost of potentially unnecessary recalculations, we could go for a cached getter pattern: Make the fields private `Option`s that cache the result of the calculation, add public `trimmed_road_line()` and `polygon()` methods that calculate on demand, and set the fields to `None` any time we change data that invalidates them.

I don't yet know if that would be an improvement, as we still need to have a clear understanding of the dependencies between the different transformations, so maybe it's something we look into more as a performance improvement later.